### PR TITLE
Updated HiFiAsm to 0.20 and changed input channels

### LINKED ISSUE #6799: Update hifiasm inputs
### Is your feature request related to a problem? Please describe

The paternal and maternal kmer inputs (as well as the HiC inputs) are specific to the reads, and I think they should be input together, i.e.:

```
tuple val(meta), path(reads), path(paternal_kmer_dump), path(maternal_kmer_dump), path(hic_read1), path(hic_read2)
```

Otherwise, I think there's no way to ensure that the reads and kmers that belong together are input to the process together, so Sample1 could get Sample2's parental kmers. 

### Describe the solution you'd like

_No response_

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by Schmytzi**
Changed HiFiAsm version to 0.20. Also joined separate input channels to one channel of tuples to make it clearer that files are associated with each other.

## PR checklist

Closes #6799 

- [x] This comment contains a description of changes (with reason).
- [x] If you've fixed a bug or added code that should be tested, add tests!
- [ ] If you've added a new tool - have you followed the module conventions in the [contribution docs](https://github.com/nf-core/modules/tree/master/.github/CONTRIBUTING.md)
- [ ] If necessary, include test data in your PR.
- [ ] Remove all TODO statements.
- [ ] Emit the `versions.yml` file.
- [ ] Follow the naming conventions.
- [x] Follow the parameters requirements.
- [x] Follow the input/output options guidelines.
- [ ] Add a resource `label`
- [x] Use BioConda and BioContainers if possible to fulfil software requirements.
- Ensure that the test works with either Docker / Singularity. Conda CI tests can be quite flaky:
  - For modules:
    - [ ] `nf-core modules test <MODULE> --profile docker`
    - [x] `nf-core modules test <MODULE> --profile singularity`
    - [ ] `nf-core modules test <MODULE> --profile conda`
  - For subworkflows:
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile docker`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile singularity`
    - [ ] `nf-core subworkflows test <SUBWORKFLOW> --profile conda`

### CODE REVIEW

File: modules/nf-core/hifiasm/main.nf [hunk 1]
```diff
@@ -4,15 +4,11 @@ process HIFIASM {
...
+        'https://depot.galaxyproject.org/singularity/hifiasm:0.20.0--h43eeafb_0' :
+        'biocontainers/hifiasm:0.20.0--h43eeafb_0' }"
 
     input:
-    tuple val(meta), path(reads)
-    path  paternal_kmer_dump
-    path  maternal_kmer_dump
-    path  hic_read1
-    path  hic_read2
+    tuple val(meta), path(reads), path(paternal_kmer_dump), path(maternal_kmer_dump), path(hic_read1), path(hic_read2)
```
Comment (muffato) [inline modules/nf-core/hifiasm/main.nf:11]: Some inputs are optional, some must co-exist, and some are mutually exclusive. Also some inputs are likely to be the same specimen as `reads`, others not.
I think it makes more sense to group maternal+paternal and the two hic like this:
```suggestion
    tuple val(meta), path(reads)
    tuple path(paternal_kmer_dump), path(maternal_kmer_dump)
    tuple path(hic_read1), path(hic_read2)
```
(I would in fact like seeing a meta map `val(metak)` added for the kmer_dump inputs, and one named `val(metah)` for the hic inputs. Even though they're not used by the module script code, having the meta maps available allows people to use their values to set resources (cpu, memory, etc) dynamically and tends to simplify code by not having to have a `map` that removes the meta map.

**Comment by mahesh-panchal** [comment]
Ah, good. I was going to ping you Matthieu.

File: modules/nf-core/hifiasm/main.nf (hunk 1 above)
Comment (mahesh-panchal) [reply modules/nf-core/hifiasm/main.nf:11]: After seeing some internal conversations today, Matthieu's comments are consistent with what we're seeing too. 
However Daniels point is that these inputs should be synchronised at some point if one is running multiple instances of hifiasm in a workflow ( which is really the same for any process input ). There was some discussion on using `multiMap` yesterday in the #team-maintainers channel, but Ben confirmed that the way we're using it is ok. 
I think I'm leaning towards this (Matthieu's) suggestion for grouping the inputs as not everyone has or needs these other inputs. I'm not going to comment on the coding ease as I have expert blindness now.

Comment (Schmytzi) [reply modules/nf-core/hifiasm/main.nf:11]: I see your point. I'll change it accordingly! 👍

**Review by mahesh-panchal — APPROVED** [review]
🚀

**Review by muffato — APPROVED** [review]

**Pull request merged**